5-RML-Reification-Statements

jhculb opened the pull request:
WIP: No work

dachafra commented on src/rml-reified/bnode-1.rml:
```diff
@@ -0,0 +1,53 @@
+        rr:reference "myFirstVariable" ;
+        rr:termType rr:BlankNode
+    ];
+    4
```
ups, and this 4?

jhculb replied:
spolling mistake

dachafra commented on src/rml-reified/bnode-1.rml:
```diff
@@ -0,0 +1,53 @@
+        rml:referenceFormulation ql:CSV
+    ];
+    rml:subjectMap [
+        rr:reference "myFirstVariable" ;
```
@andimou how do we manage this? As we shouldn't have this reference for the reification standard blank node in the data... Is there any automatic generation for BN in RML?

jhculb commented:
Good evening @dachafra @andimou, hope you are well. Could you review and approve the merge when you are next free please!

jhculb commented on src/rml-reified/nested-1.ttl:
```diff
@@ -0,0 +1,91 @@
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix ql: <http://semweb.mmlab.be/ns/ql#> .
+@prefix ex: <http://example/> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#>
```
I realise these are extraneous prefixes. I assume it's fine keeping them around

dachafra replied:
this is good!

dachafra commented on src/rml-reified/syntax-2.ttl:
```diff
@@ -0,0 +1,83 @@
+            rr:template "http://example/{c3}"
+        ]
+    ];
+
```
did you validate this RDF? I think you need a `.` at the end

dachafra replied:
review in the rest please

jhculb replied:
> did you validate this RDF? I think you need a `.` at the end

I did when parsing the triples as part of 6.

dachafra commented on src/rml-reified/syntax-3.ttl:
```diff
@@ -0,0 +1,121 @@
+    rr:predicateObjectMap [
+        rr:predicate rdf:subject ;
+        rml:objectMap [
+            rml:quotedTriplesMap :elementaryTM1
```
NO! You cannot use this in plan RML, there should be another way ;-)

dachafra commented on src/rml-reified/syntax-3.ttl:
```diff
@@ -0,0 +1,121 @@
+    rr:predicateObjectMap [
+        rr:predicate rdf:object ;
+        rml:objectMap [
+            rml:quotedTriplesMap :elementaryTM2
```
again, this is not good

dachafra commented on src/rml-reified/syntax-4.ttl:
```diff
@@ -0,0 +1,121 @@
+    rr:predicateObjectMap [
+        rr:predicate rdf:object ;
+        rml:objectMap [
+            rml:quotedTriplesMap :elementaryTM2
```
again, we cannot use this!

dachafra reviewed (changes requested):
You cannot use rml:quotedTriplesMaps in plan RDF, you may need to take a look to RML spec to see how you can do it (maybe just repeating some rules...)

jhculb commented:
> You cannot use rml:quotedTriplesMaps in plan RDF, you may need to take a look to RML spec to see how you can do it (maybe just repeating some rules...)

Are you referring to parentTriplesMap? This seems to be how to map two logical sources (csvs, etc.) together?

(currently reading R2RML spec to see if I've missed something there)

dachafra commented:
It could be a good option, yes ;-)

jhculb commented:
> It could be a good option, yes ;-)

So I think it is a direct replacement in all occurences in these examples, though reading the rml.io specs was a little unclear about parent and children, and I feel I may have trippled up here.

Could you give this a review please as I think I have addressed your comments?

dachafra commented:
If the source of the parentTriplesMap and the actual TriplesMap is the same, you do not need to put a join-condition

dachafra reviewed (approved):
ready to merge
